projects/common/src/lib/components/iot-pipeline/iot-pipeline.component.ts

```diff
@@ -40,3374 +40,3529 @@ import { RestBroadcastComponent } from '../rete/components/rest-broadcast.compon
+                },
+                {
+                    'name': 'deviceName',
+                    'type': 'String'
+                },
+                {
+                    'name': 'resourceName',
+                    'type': 'String'
+                },
+                {
+                    'name': 'value',
+                    'type': 'string'
+                }
+            ]
+        }
+    ]
+
+
+    private filterComponent: PipelineFilteringComponent;
+
+    @ViewChild('filterComponent') set fcontent(content: PipelineFilteringComponent) {
+        if (content) { // initially setter gets called with undefined
+            this.filterComponent = content;
+        }
+    }
+
+    private inferenceComponent: PipelineInferencingComponent;
+    @ViewChild('inferenceComponent') set icontent(content: PipelineInferencingComponent) {
+        if (content) { // initially setter gets called with undefined
+            this.inferenceComponent = content;
+        }
+    }
+
+    private restComponent: PipelineRestServiceComponent;
+    @ViewChild('restComponent') set rcontent(content: PipelineRestServiceComponent) {
+        if (content) { // initially setter gets called with undefined
+            this.restComponent = content;
+        }
+    }
+
+    @ViewChild('nodeEditor') el: ElementRef;
+    editor = null;
+
+    constructor(private graphService: GraphService,
+        private edgeService: EdgeService,
+        private flogoDeployService: FlogoDeployService,
+        private route: ActivatedRoute,
+        private formBuilder: FormBuilder,
+        private _snackBar: MatSnackBar) {
+
+    }
+
+    ngOnInit(): void {
+        // Get gateway id from route
+        this.gatewayId = this.route.snapshot.paramMap.get('gatewayId');
+
+        this.getGatewayDetails(this.gatewayId);
+
+        this.createForms();
+    }
+
+
+    async ngAfterViewInit() {
+
+        console.log('PipelineEditor - ngAfterViewInit - devices: ', this.devices);
+
+        const container = this.el.nativeElement;
+
+        // new ImageResizeComponent();
+
+        const components = [
+            new DataSubscriberComponent(),
+            new DataPublisherComponent(),
+            new DataStoreComponent(),
+            new FiltersComponent(),
+            new InferencingComponent(),
+            new RulesComponent(),
+            new StreamingComponent(),
+            new RuleExpressionComponent(),
+            new RuleTextMatchingComponent(),
+            new RestServiceComponent(),
+            new RestBroadcastComponent(),
+            new MQTTRequestReplyComponent(),
+            new NotificationPublisherComponent(),
+            new ImageResizeComponent(),
+            new CustomPublisherComponent(),
+            new FlogoFlowComponent(),
+            new ErrorHandlerComponent(),
+            new PythonComponent(),
+            new PythonPipeComponent()
+        ];
+
+        this.editor = new NodeEditor('demo@0.2.0', container);
```

Comment:
What is `demo@0.2.0`? Will this value ever change or be shown?

Reply:
Not shown. It's just the name of the `rete` editor if I'm not mistaken.